oshi-core/src/main/java/oshi/util/ExecutingCommand.java

```diff
@@ -116,11 +116,33 @@ private ExecutingCommand() {
      *         string if the command failed
      */
     public static List<String> runNative(String[] cmdToRunWithArgs, String[] envp) {
+        Process p = null;
         try {
-            Process p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
+            p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
             return getProcessOutputAndDestroy(p, cmdToRunWithArgs);
         } catch (SecurityException | IOException e) {
             LOG.trace("Couldn't run command {}: {}", Arrays.toString(cmdToRunWithArgs), e.getMessage());
+        } finally {
+            if (p != null) {
```

Comment:
If `exec()` fails with an exception, we will never execute this block.

Reply:
Hi Daniel,
You wrote "if you want to move the finally block from getProcessOutputAndDestroy() to this method (and change the name)...".
I did not understand what method and what name you are referring to.
10x,
David

Reply:
> I did not understand what method and what name you are referring to.

A few weeks ago, I noted the case you're referring to in this PR, where `exec()` fails.  The PR #1767 was intended to address this case.  My solution in that case was to add a `finally` block in the method `getProcessOutputAndDestroy()` which terminated the process (and closed all its resources).

If you think the `p.destroy()` call is better placed in a `finally` block in the `runNative()` method you've addressed in your PR, I'm okay with that.  The point I'm making is that the `destroy()` method accomplishes all the stream closures that you are proposing in this PR.

Regarding the "change the name" I intentionally added "AndDestroy" to the name to make it clear that this method call would kill the process after collecting its output.  If you're moving the `destroy()` call out of that method then its name should just be `getProcessOutput()`.